src/lib/krb5/ccache/cc_retr.c

```diff
@@ -162,41 +159,52 @@ pref (krb5_enctype my_ktype, int nktypes, krb5_enctype *ktypes)
  */
 
 krb5_boolean
-krb5int_cc_creds_match_request(krb5_context context, krb5_flags whichfields, krb5_creds *mcreds, krb5_creds *creds)
+krb5int_cc_creds_match_request(krb5_context context, krb5_flags whichfields,
+                               krb5_creds *mcreds, krb5_creds *creds)
 {
-    /* Only match a user-to-user credential if explicitly asked for. */
-    if (! set(KRB5_TC_MATCH_IS_SKEY) && creds->is_skey)
+    krb5_boolean is_skey;
+
+    if ((whichfields & KRB5_TC_MATCH_SRV_NAMEONLY)) {
+        if (!srvname_match(context, mcreds, creds))
+            return FALSE;
+    } else {
+        if (!standard_fields_match(context, mcreds, creds))
+            return FALSE;
+    }
+
+    /* Only match a user-to-user credential if explicitly asked for, since the
+     * ticket won't work as a regular service ticket. */
```

Comment:
This comment is added in the first commit, but gets expanded in the third one.  Probably should just be added wholesale in the first?